src/js/mediasession.js

```diff
@@ -0,0 +1,132 @@
+      this.pause();
+    }],
+    ['stop', () => {
+      this.pause();
+      this.currentTime(0);
+    }],
+    // videojs-contrib-ads
+    ['seekbackward', (details) => {
+      if (this.usingPlugin('ads') && this.ads.inAdBreak()) {
+        return;
+      }
+      this.currentTime(Math.max(0, this.currentTime() - (details.skipOffset || defaultSkipTime)));
+    }],
+    ['seekforward', (details) => {
+      if (this.usingPlugin('ads') && this.ads.inAdBreak()) {
+        return;
+      }
+      this.currentTime(Math.min(this.duration(), this.currentTime() + (details.skipOffset || defaultSkipTime)));
+    }],
+    ['seekto', (details) => {
+      if (this.usingPlugin('ads') && this.ads.inAdBreak()) {
+        return;
+      }
+      this.currentTime(details.seekTime);
+    }]
+  ];
+
+  // Using Google's recommendation that expects some handler may not be settable, especially as we
+  // want to support older Chrome
+  // https://web.dev/media-session/#let-users-control-whats-playing
+  for (const [action, handler] of actionHandlers) {
+    try {
+      ms.setActionHandler(action, handler);
+    } catch (error) {
+      this.log.debug(`Couldn't register media session action "${action}".`);
+    }
+  }
+
+  const setUpMediaSessionPlaylist = () => {
+    try {
+      ms.setActionHandler('previoustrack', () => {
+        this.playlist.previous();
+      });
+      ms.setActionHandler('nexttrack', () => {
+        this.playlist.next();
+      });
+    } catch (error) {
+      this.log.debug('Couldn\'t register playlist media session actions.');
+    }
+  };
+
+  // Only setup playlist handlers if / when playlist plugin is present
+  this.on('pluginsetup:playlist', setUpMediaSessionPlaylist);
+
+  /**
+   *
+   * Updates the mediaSession metadata. Fires `updatemediasession` as an
+   * opportunity to modify the metadata
+   *
+   * @fires Player#updatemediasession
+   */
+  const updateMediaSession = () => {
+    const currentMedia = this.getMedia();
+    const playlistItem = this.usingPlugin('playlist') ? Object.assign({}, this.playlist()[this.playlist.currentItem()]) : {};
+    const mediaSessionData = {
+      title: currentMedia.title || playlistItem.name || '',
+      artist: currentMedia.artist || playlistItem.artist || '',
+      album: currentMedia.album || playlistItem.album || ''
+    };
+
+    if (currentMedia.artwork) {
+      mediaSessionData.artwork = currentMedia.artwork;
+    } else if (playlistItem.artwork) {
+      mediaSessionData.artwork = playlistItem.artwork;
+    } else if (this.poster()) {
+      mediaSessionData.artwork = [{
+        src: this.poster(),
+        type: getMimetype(this.poster())
+      }];
+    }
+
+    // This allows the metadata to be updated before being set, e.g. if loadMedia() is not used.
+    this.trigger('updatemediasession', mediaSessionData);
+
+    ms.metadata = new window.MediaMetadata(mediaSessionData);
+  };
+
+  const updatePositionState = () => {
+    const dur = parseFloat(this.duration());
```

Comment:
Question, is there a case that requires a `parseFloat`?

Reply:
That can probably go

Reply:
From a specification perspective ([w3c](https://www.w3.org/TR/2011/WD-html5-20110113/video.html#dom-media-duration), [whatwg](https://html.spec.whatwg.org/multipage/media.html#dom-media-duration-dev)), it seems safe to remove the `parseFloat`, especially as the condition immediately following it prevents any `NaN` or `Infinity`.  However, if you prefer to keep it, that's fine too.